WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -26,10 +28,15 @@ namespace WalletWasabi.Fluent.ViewModels.CoinControl;
 	NavigationTarget = NavigationTarget.DialogScreen)]
 public partial class SelectCoinsDialogViewModel : DialogViewModelBase<IEnumerable<SmartCoin>>
 {
-	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel)
+	private readonly CompositeDisposable _disposables = new ();
+
+	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel, IEnumerable<SmartCoin> selectedCoins)
 	{
 		var pockets = walletViewModel.Wallet.GetPockets();
 		var items = CreateItems(pockets);
+		items.ToList().ForEach(x => x.DisposeWith(_disposables));
```

Comment:
I don't know what to do with this, just doesn't look good that it is converted (again) to a list just to have a `ForEach` call.

Using `SourceList` would be overkill? To have `.DisposeMany()`.
Or you can simply iterate through the Source.Items and call dispose when navigating away?

Reply:
I'm using your second suggestion. This could be done more elegantly, but this is more than enough.